illumos-utils/src/zfs.rs

```diff
@@ -1339,6 +1348,39 @@ impl Zfs {
         }
         Ok(result)
     }
+
+    pub async fn ensure_dataset_volume(
+        name: String,
+        size: ByteCount,
+        block_size: u32,
```

Comment:
Same question about newtypes - is any `u32` valid for a block size, or should this be a `BlockSize(u32)` (or even a `BlockSize::*` enum if there are only a few choices that we allow)?